File: Python/lc_1_two_sum.py
# Time: O(n), n=len(nums)
# Space: O(n)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        imap = {}

        for index, num in enumerate(nums):
            if target-num in imap:
                return [imap[target-num], index]
            else:
                imap[num] = index

        return [None, None]

File: Go/lc_1_two_sum.go

package go

// Time: O(n)
// Space: O(n)

func twoSum(nums []int, target int) []int {

	imap := make(map[int]int)

	for index1, val1 := range nums {
		if index2, ok := imap[target-val1]; ok {
			return []int{index2, index1}
		}

		imap[val1] = index1
	}

	return []int{-1, -1}
}
